configure.ac

```diff
@@ -8,6 +8,37 @@ AC_CONFIG_HEADERS([src/config.h])
 AM_INIT_AUTOMAKE([foreign dist-xz -Wall -Werror])
 AC_CONFIG_MACRO_DIRS([m4])
 
+# Check whether we have a p11-kit to use as a default PKCS#11 module
+AC_ARG_WITH([default-pkcs11-module],
+       [ --with-default-pkcs11-module[[=PATH]]   Use default PKCS#11 module (p11-kit detected by default)],
+       [ if test "x$withval" != "xno"; then
+               if test "x$withval" = "xyes" ; then
+                       AC_PATH_TOOL([PKGCONFIG], [pkg-config], [no])
+                       if test "x$PKGCONFIG" != "xno"; then
+                               AC_MSG_CHECKING([if $PKGCONFIG knows about p11-kit])
+                               if "$PKGCONFIG" "p11-kit-1"; then
+                                       AC_MSG_RESULT([yes])
+                                       use_pkgconfig_for_p11kit=yes
+                               else
+                                       AC_MSG_RESULT([no])
+                               fi
+                       fi
+               else
+                       PKCS11_PATH="${withval}"
+               fi
+               if test "x$use_pkgconfig_for_p11kit" = "xyes"; then
+                       PKCS11_PATH=`$PKGCONFIG --variable=proxy_module p11-kit-1`
+               fi
+               AC_CHECK_FILE("$PKCS11_PATH",
+                       [ AC_DEFINE_UNQUOTED([DEFAULT_PKCS11_MODULE],
+                                            ["$PKCS11_PATH"],
+                                            [Path to default PKCS#11 module])
+                       ],
+                       [ AC_MSG_ERROR([Requested PKCS11 provided not found]) ]
+               )
+       fi ]
+)
+
```

Comment:
Looks over complicated, what is wrong in using directly something like:
```
PKG_CHECK_EXISTS(
        [p11-kit-1],
        [PKG_CHECK_VAR([...
```

See how this is done for nss-sotoken please.

Reply:
I wanted to give a chance to enable/disable this, but I think this should be elected by the build system by providing p11-kit or not so lets simplify this.